scripts/examples/typescript/Makefile

```diff
@@ -0,0 +1,12 @@
+.PHONY: build clean
+
+build:
+	sed -i '/version/s/1.0.0/'"$(XAPI_VERSION)"'/' package.json
+	npm install
+	npm run build
+
+publish:
+	npm publish
```

Comment:
honestly don't think we'd need this. why are we adding a `publish` here?

Reply:
futhermore, in `npm` you'd usually use scripts in the `package.json`, not `Makefile`s. I'd move everything there if you need it

for instance, your expanded build could be `build:production` under `scripts`

Reply:
I think the publish might be useful if we want to publish automatically to npm like we do for XenAPI.py on pypi?

Reply:
> I think the publish might be useful if we want to publish automatically to npm like we do for XenAPI.py on pypi?

Yes I agree but we're not publishing right now. Unless that's planned with this PR, in which case we need GitHub actions, and either a publishing setup on GitHub or npmjs.com.

Also for publishing I'd argue we'd need to set up a XenServer account. Let's do it properly. For instance I don't see why we have this already: https://www.npmjs.com/package/xen-api-ts